### ISSUE TITLE
Module '"redux-saga/effects"' has no exported member 'call'.

### ISSUE THREAD

**Issue by tsopeh** [redux-saga/redux-saga#2389]
### Steps to reproduce

Try importing the `call` (or any other) method in a .js or .ts file.

#### Description of the bug/issue

There is no code completion, suggestions or documentation in WebStorm and StackBlitz.

For .ts files IDE says: `Module '"redux-saga/effects"' has no exported member 'call'.`

#### Steps to reproduce the bug/issue

1. Open this [StackBlitz](https://stackblitz.com/edit/typescript-ojxbva?file=index.ts,index-js.js) repro
2. Take a look at the `index.ts` and `index-js.js`

#### Example

```typescript
import { call } from 'redux-saga/effects';
```

### Actual results

IDE: `Module '"redux-saga/effects"' has no exported member 'call'.`

### The Expected results

Successfully resolved import.

### Environment information

- redux-saga version - 1.2.3
- babel / typescript setup - TS 4.9.3

### Note

I've tried to bring the attention to this issue in the following issues, but got no answers for two weeks:
- https://github.com/redux-saga/redux-saga/issues/1931#issuecomment-1695650937
- https://github.com/redux-saga/redux-saga/issues/2013#issuecomment-1695811875

**Comment by neurosnap** [comment]
This must be an issue with StackBlitz because when I download the project locally and run it, there are no typescript issues.  Could you please create a github repo that I could clone locally that replicates this issue?

**Comment by tsopeh** [comment]
It works well in VS Code (locally), but JS files in WebStorm (locally) do not recognise types for "redux-saga/effects". Many other libraries work fine.

Here's the exported StackBlitz project. You can export it as well by clicking to the "Download Project" button.

ZIP: [typescript-ojxbva.zip](https://github.com/redux-saga/redux-saga/files/12594595/typescript-ojxbva.zip)

![image](https://github.com/redux-saga/redux-saga/assets/37545996/e6b5cc3d-e112-44af-94e6-dd15e13a1593)

**Comment by neurosnap** [comment]
Running that project using `vite` works fine.  Running the project using `VS Code` works fine.  Is it possible this is an issue specifically with `WebStorm`?  Are there any other IDEs or bundlers that you notice the same typing issue?

I don't want to be dismissive and say this is a `WebStorm` problem so I'm happy to guide anyone through this issue and accept any PRs that fix the problem.  However, I don't personally have the bandwidth to debug an IDE specific issue with `redux-saga`.

Thanks for raising this issue and again, happy to help where I can on fixing this issue.

**Comment by simlmx** [comment]
I had the same issue using `rollup` and this typescript plugin: https://www.npmjs.com/package/rollup-plugin-typescript2

Using the official plugin instead fixed the issue:
https://www.npmjs.com/package/@rollup/plugin-typescript

**Comment by Skladnayazebra** [comment]
I googled my way here, had the same issue with my project. After some tinkering, I think I found the culprit.

**TL;DR:** in my case, this error was caused by `"preserveSymlinks": true` tsconfig option when using pnpm.

First I checked if it works on a fresh new project.
I started an empty project using **pnpm**, installed latest **typescript** (5.9.3) and **redux-saga** (1.4.2) packages, and some boilerplate code:

```typescript
import { put } from 'redux-saga/effects'

function* mySaga () {
	yield put({type: "actiontype" })
}
```
and a minimal tsconfig.json
```json
{
	"compilerOptions": {
		"outDir": "dist",
		"rootDir": "src",
		"lib": ["DOM.Iterable", "DOM","ESNext"],
	}
}
```
With this setup, LSP in my editor doesn't show any errors and `npx tsc` compiles with no errors, which proves that the library works with default setup.

After that, I went back to my broken project and started removing compiler options one by one, running `npx tsc` after each removal. When I removed `"preserveSymlinks": true,`, the error disappeared.

Then I went back to the boilerplate project and added that option to tsconfig.
```json
{
	"compilerOptions": {
		"outDir": "dist",
		"rootDir": "src",
		"lib": ["DOM.Iterable", "DOM","ESNext"],
		"preserveSymlinks": true, // new option
	}
}
```
After that, I saw the error in editor and `npx tsc` failed as well.

As [pnpm docs](https://pnpm.io/typescript#do-not-preserve-symlinks) states:
> You should not use TypeScript with `preserveSymlinks` set to `true`.

Maybe that was the case for OP as well?
